src/components/Interactive.js

```diff
@@ -0,0 +1,65 @@
+import _ from 'lodash/fp'
+import { forwardRef, useState } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import * as Utils from 'src/libs/utils'
+
+
+const allowedHoverVariables = ['backgroundColor', 'color', 'boxShadow', 'opacity', 'textDecoration']
+const pointerTags = ['button', 'area', 'a', 'select']
+const pointerTypes = ['radio', 'checkbox', 'submit', 'button']
+
+const Interactive = forwardRef(({
+  className = '', as, type, role, onClick, onKeyDown, onMouseDown, onBlur, disabled, children, tabIndex, hover = {}, style = {}, ...props
+}, ref) => {
+  const [outline, setOutline] = useState(undefined)
+  const { cursor } = style
+
+  const computedCursor = Utils.cond(
+    [cursor, cursor],
+    [disabled, undefined],
+    [onClick || pointerTags.includes(as) || pointerTypes.includes(type), 'pointer']
+  )
+
+  const computedTabIndex = Utils.cond(
+    [_.isNumber(tabIndex), tabIndex],
+    [disabled, undefined],
+    [onClick, 0],
+    undefined)
+
+  const computedRole = Utils.cond(
+    [role, role],
+    [onClick && !['input', ...pointerTags].includes(as), 'button'],
+    undefined)
+
+  const cssVariables = _.flow(
+    _.toPairs,
+    _.flatMap(([key, value]) => {
+      console.assert(allowedHoverVariables.includes(key),
+        `${key} needs to be added to the hover-style in style.css for the style to be applied`)
+      return [[`--app-hover-${key}`, value], [key, `var(--hover-${key}, ${style[key]})`]]
+    }),
+    _.fromPairs
+  )(hover)
+
+  return h(as, {
+    ref,
+    className: `hover-style ${className}`,
+    style: { ...style, ...cssVariables, cursor: computedCursor, outline },
+    onKeyDown: onKeyDown || (e => e.key === 'Enter' && _.forEach(fn => fn(), [() => e.stopPropagation(), () => e.target.click()])),
```

Comment:
@panentheos I believe this resolves the issue with the click event bubbling.
Two questions:
1. This does mean anyone passing in an onKeyDown handler will have to add a stopPropagation on enter keypresses to their keydown handler if it is in a nested interactive AND they do not want a click bubbling. If they do not add the stopPropagation when the user hits 'enter' the keydown event will bubble up to the parent Interactive, triggering a click (if the parent interactive does not have a keydown handler). This is a specific case and it could be something that anyone consuming interactive will need to be aware of. I want to call it out to make sure there are no issues with this. There are 3 keyDown handlers in the code base on searchInput, confirmedSearchInput and in NewUserModal.

2. Thoughts on using the forEach to inline a few function calls dependent on the same condition rather than creating a function with an explicit conditional - is this clear in the code, simpler, etc...

Reply:
1. I think that's fine, having nested Interactives is rare enough, never mind with custom key handling. We can always tweak the behavior if we run into it and it makes things awkward.

2. Since we're doing imperative things here, I think just a multiline function with an `if` would be clearer.